src/javascript/app/pages/user/account/settings/personal_details.js

```diff
@@ -362,16 +471,16 @@ const PersonalDetails = (() => {
     );
 
     const onLoad = () => {
-        BinarySocket.wait('get_account_status', 'get_settings').then(() => {
+        BinarySocket.wait('get_account_status', 'get_settings', 'landing_company').then(() => {
             init();
             const account_status = State.getResponse('get_account_status').status;
             get_settings_data = State.getResponse('get_settings');
             is_fully_authenticated = checkStatus(account_status , 'authenticated');
             if (is_virtual) {
                 getDetailsResponse(get_settings_data);
             }
-
-            if (!is_virtual || !residence) {
+            if (!is_virtual || !residence || has_changeable_fields) {
```

Comment:
will there every be a case where client is virtual or doesn't have residence but has changeable fields? I think no, if client is virtual there are no changeable fields because we only allow CR to do that, and if client doesn't have residence the first thing they have to do is select residence before the rest of their details is displayed to them.

Reply:
I'll refactor this to separate the previous behavior from injecting `has_changeable_field`